File: AbstractFactoryUsingBuilderGOlang/abstractFactory/HAWAIIANDealer.go
package abstractFactory

import (
	"myProject7/PizzaBuilder"
	"myProject7/SandwichBuilder"
)

type HAWAIIANDealer struct{}

func (V HAWAIIANDealer) CreatePizza() PizzaBuilder.Pizza {

	hawaiianBuilder := PizzaBuilder.GetBuilder("hawaiian")
	waitress := PizzaBuilder.NewWaitressPizza(hawaiianBuilder)
	hawaiianPizza := waitress.BuildPizza()

	return hawaiianPizza
}

func (V HAWAIIANDealer) CreateSandwich() SandwichBuilder.Sandwich {
	hawaiianBuilder := SandwichBuilder.GetBuilder("hawaiian")
	waitress := SandwichBuilder.NewWaitressSandwich(hawaiianBuilder)
	hawaiianSandwich := waitress.BuildSandwich()

	return hawaiianSandwich
}

File: AbstractFactoryUsingBuilder/AbstractFactory/AbstractFactory/HAWAIIANDealer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstractFactory
{
    public class HAWAIIANDealer : AbstractFactory
    {

        public override PIZZA createPIZZA()
        {
            PizzaBuilder builderHAWAIIANPIZZA = new BuilderHAWAIIANPIZZA();
            WAITRESSPIZZA wai = new WAITRESSPIZZA(builderHAWAIIANPIZZA);
            wai.constructPizza();

            return wai.getPizza();
        }

        public override SANDWICH createSANDWICH()
        {
            SandwichBuilder builderHAWAIIANSANDWICH = new BuilderVEGGYSANDWICH();
            WAITRESSSANDWICH wai = new WAITRESSSANDWICH(builderHAWAIIANSANDWICH);
            wai.constructSandwich();

            return wai.getSandwich();
        }

    }
}
